.github/workflows/actions/publish-npm.yml

```
@@ -0,0 +1,44 @@
+name: 'Release'
+description: 'Releases a package'
+inputs:
+  scope:
+    description: 'The package to release. Must match a package specified in lerna.json.'
+  version:
+    description: 'The type of version to release.'
+  tag:
+    description: 'The tag to publish to on NPM (DRY RUN).'
+  preid:
+    description: 'The prerelease identifier used when doing a prerelease.'
+  working-directory:
+    required: true
+    description: 'The directory of the package to deploy.'
+  token:
+    description: 'The NPM authentication token required to publish.'
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/setup-node@v3
```

Can we use the latest version of this action? ATM, `actions/checkout` is on v3. (3.5.2, specifically):

```suggestion
    - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```

---

I have to touch all of these in the next PR I have anyway, let's let these ride for now and I'll fix the specific pinning of all sha's. This is an interesting blog post and workflow https://michaelheap.com/ensure-github-actions-pinned-sha/ that would be nice-to-have.